src/components/PeoplePages/PeoplePages.tsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-param-reassign */
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { Person } from '../../types/Person';
+import { PeopleTable } from '../PeopleTable';
+import { useParams } from 'react-router-dom';
+
+const setParent = (personArr: Person[]) => {
+  personArr.map(person => {
+    person.father = personArr.find(
+      personFather => person.fatherName === personFather.name,
+    );
+    person.mother = personArr.find(
+      personMother => person.motherName === personMother.name,
+    );
+  });
+};
+
+export const PeoplePages = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [erroreMessage, setError] = useState('');
+  const { index } = useParams();
+  const selectedPerson = index ? index : '';
+
+  useEffect(() => {
+    getPeople()
+      .then(setPeoples)
+      .catch(errore => {
+        setError('Something went wrong');
+        throw errore;
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    setParent(peoples);
+    if (peoples.length) {
+      setError('There are no people on the server');
+    }
+  }, [peoples]);
```

Comment:
I have no idea why you do it.
You can just prepare an array when you get from the API and set it with `setPeople`